src/main/java/com/firebolt/jdbc/client/query/StatementClientImpl.java

```diff
@@ -334,6 +328,18 @@ private Map<String, String> getAllParameters(FireboltProperties fireboltProperti
 		return params;
 	}
 
+	/**
+	 * Each statement will have a query label (most of them will be autogenerated except for the SET query_label explicit statement which will use the specified query label).
+	 * When encountering a SET query_label statement we will set it on the firebolt properties and use that one instead the one created on the statementInfoWrapper
+	 * @param fireboltProperties
+	 * @param statementInfoWrapper
+	 * @return
+	 */
+	private String determineQueryLabel(FireboltProperties fireboltProperties, StatementInfoWrapper statementInfoWrapper) {
+		String existingQueryLabel = fireboltProperties.getRuntimeAdditionalProperties().get(FireboltQueryParameterKey.QUERY_LABEL.getKey());
+		return StringUtils.isNotBlank(existingQueryLabel) ? existingQueryLabel : statementInfoWrapper.getLabel();
+	}
```

Comment:
Looks like we have several places where we have essentially the same logic. Can we unify this somehow? We're not accessing attributes of this class so this doesn't have to be a method here. We can have it anywhere we want and we can make it static.